README.md

```diff
@@ -446,6 +446,23 @@ end
 
 See the section on *Script fields* for details on calculating distance in a search.
 
+### Join fields
+
+You can use a [join field](https://www.elastic.co/guide/en/elasticsearch/reference/current/parent-join.html)
+to implement parent-child relationships between documents.
+It [replaces the old `parent_id` based parent-child mapping](https://www.elastic.co/guide/en/elasticsearch/reference/current/removal-of-types.html#parent-child-mapping-types)
+
+To use it, you need to pass `relations`, `join_type` and `join_id` options:
+```ruby
+field :hierarchy_link, type: :join, relations: {question: %i[answer comment], answer: :vote, vote: :subvote}, join_type: :comment_type, join_id: :commented_id
```

Comment:
Not a blocker, but what do you think about make the syntax less verbose, like:

```suggestion
field :hierarchy_link, join: { type: :comment_type, id: :commented_id }, relations: {question: %i[answer comment], answer: :vote, vote: :subvote}
```

So I would drop the requirement of declaring `type: :join` once we have a join attribute with `type` and `id` instead of the pair `join_type` and `join_id`

Reply:
I like the idea to make the options simpler, yet I'm not sure about skipping the `type:` key. It's aligned with other types, e.g.
```
field :created, type: 'date', include_in_all: false,
    value: ->{ created_at } # value proc for source object context
end
```
WDYT about keeping `type: join`, but moving `relations`, `id` and `type` to `join:` key, i.e.

```suggestion
field :hierarchy_link, type: :join, join: { relations: {question: %i[answer comment], answer: :vote, vote: :subvote}, type: :comment_type, id: :commented_id }
```

Reply:
I like the suggestion of grouping `join_type` and `join_id` inside join.

Also my suggestion to drop type wasan't to actually drop it, but making it not mandatory if we have defined join keyword... But anyway, enforcing it or not, it is good to me

Reply:
Finally I kept relations outside of the `join` param to make it aligned with actual elastic syntax (so we can just skip the `join` option when passing to elastic).